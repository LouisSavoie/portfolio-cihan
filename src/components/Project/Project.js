import React from 'react'
import Card from 'react-bootstrap/Card'
import ListGroup from 'react-bootstrap/ListGroup'
import Button from 'react-bootstrap/Button'

const Project = ({ project }) => {
  const { name, picture, tech, about, repo, demo } = project

  const displayTech = () => {
    return tech.map(techItem => {
      return <ListGroup.Item key={techItem}>{techItem}</ListGroup.Item>
    })
  }

  const displayButtons = (dataValue, text) => {
    return dataValue !== '' && <Button variant="primary" size="lg" href={dataValue} target="_blank" rel="noopener noreferrer">{text}</Button>
  }

  return (
    <div data-testid="project">
      <Card style={{ width: '18rem' }}>
        <Card.Img variant="top" src={picture} />
        <Card.Body>
          <Card.Title>{name}</Card.Title>
          <ListGroup>
            {displayTech()}
          </ListGroup>
          <Card.Text>{about}</Card.Text>
          {displayButtons(repo, 'Repo')}
          {displayButtons(demo, 'Demo')}
        </Card.Body>
      </Card>
    </div>
  )
}

export default Project
