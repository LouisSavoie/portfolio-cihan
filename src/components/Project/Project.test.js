import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import Project from './Project'

const project = {
  name: 'Bacon',
  picture: './img/bacon.png',
  tech: [
    'Short ribs',
    'chuck',
    'pastrami',
    'corned beef',
    'chicken'
  ],
  about: 'Bacon ipsum dolor amet chislic ribeye drumstick buffalo jowl beef ribs cow bresaola pig brisket hamburger. Andouille picanha jerky cupim ground round filet mignon tenderloin sirloin venison ribeye turducken. Andouille tenderloin ground round pastrami rump pork chop, sausage buffalo porchetta kevin pancetta jerky hamburger. Shank pork beef, spare ribs tri-tip pork belly doner beef ribs strip steak meatball hamburger short loin jerky. Pork rump meatloaf frankfurter cow short loin drumstick venison tenderloin tail beef ribs. Ham alcatra chuck, boudin burgdoggen t-bone beef cow doner pastrami shoulder chicken drumstick capicola spare ribs.',
  repo: 'https://github.com/petenelson/bacon-ipsum',
  demo: 'https://baconipsum.com/'
}

afterEach(() => {
  cleanup()
})

test('should render Project', () => {
  render(<Project key={project.name} project={project} />)
  const ProjectElement = screen.getByTestId('project')
  expect(ProjectElement).toBeInTheDocument()
})

test('matches snapshot', () => {
  const tree = renderer.create(<Project key={project.name} project={project} />).toJSON()
  expect(tree).toMatchSnapshot()
})
