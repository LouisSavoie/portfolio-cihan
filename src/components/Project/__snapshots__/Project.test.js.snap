// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="project"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
      src="./img/bacon.png"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Bacon
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          Short ribs
        </div>
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          chuck
        </div>
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          pastrami
        </div>
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          corned beef
        </div>
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          chicken
        </div>
      </div>
      <p
        className="card-text"
      >
        Bacon ipsum dolor amet chislic ribeye drumstick buffalo jowl beef ribs cow bresaola pig brisket hamburger. Andouille picanha jerky cupim ground round filet mignon tenderloin sirloin venison ribeye turducken. Andouille tenderloin ground round pastrami rump pork chop, sausage buffalo porchetta kevin pancetta jerky hamburger. Shank pork beef, spare ribs tri-tip pork belly doner beef ribs strip steak meatball hamburger short loin jerky. Pork rump meatloaf frankfurter cow short loin drumstick venison tenderloin tail beef ribs. Ham alcatra chuck, boudin burgdoggen t-bone beef cow doner pastrami shoulder chicken drumstick capicola spare ribs.
      </p>
      <a
        className="btn btn-primary btn-lg"
        disabled={false}
        href="https://github.com/petenelson/bacon-ipsum"
        onClick={[Function]}
        onKeyDown={[Function]}
        rel="noopener noreferrer"
        role="button"
        tabIndex={0}
        target="_blank"
      >
        Repo
      </a>
      <a
        className="btn btn-primary btn-lg"
        disabled={false}
        href="https://baconipsum.com/"
        onClick={[Function]}
        onKeyDown={[Function]}
        rel="noopener noreferrer"
        role="button"
        tabIndex={0}
        target="_blank"
      >
        Demo
      </a>
    </div>
  </div>
</div>
`;
